control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -142,6 +145,11 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error
 		return fmt.Errorf("failed setting up with a controller manager %w", err)
 	}
 
+	// check for scc capability
+	if !r.ManagementClusterCapabilities.Has(capabilities.CapabilitySecurityContextConstraint) {
+		r.SetSecurityContext = true
```

Comment:
why do you need another field in the struct? Seems to me`r.ManagementClusterCapabilities.Has(capabilities.CapabilitySecurityContextConstraint)` is already the bool you need.

Reply:
We can surely do that, but it seemed cleaner to set/update another field. It's used throughout the file.

We can either leave it here or move it to `main.go` and update the struct field from there.